crates/cranelift/src/compiler.rs

```diff
@@ -1002,6 +1002,12 @@ impl FunctionCompiler<'_> {
             path.set_extension("clif");
 
             let mut output = std::fs::File::create(path).unwrap();
+            write!(
+                output,
+                ";;Intermediate Representation of function <{}>:\n",
+                symbol.split("::").last().unwrap()
```

Comment:
This would work too, right?

```suggestion
                ";; Intermediate Representation of function <{symbol}>:\n",
```

This avoids truncating a function name like `wasm[0]::function[1]::my_library::SomeType::get` (as would be used if the names section contains `my_library::SomeType::get` as name) to the pretty much useless `get`.

Reply:
would using something like this function work better?

```rust
fn remove_first_two_segments(s: &str) -> Option<String> {
    let mut parts = s.split("::").skip(2);
    let rest: Vec<&str> = parts.collect();
    if rest.is_empty() {
        None
    } else {
        Some(format!("::{}", rest.join("::")))
    }
}
```

also wasmtime explore uses "Intermediate Representation of function <get>:" to should i figure out a way where it gives my_library::SomeType::get too?